crates/oxc_semantic/src/checker/typescript.rs

```diff
@@ -542,3 +542,32 @@ pub fn check_for_statement_left(left: &ForStatementLeft, is_for_in: bool, ctx: &
         }
     }
 }
+
+fn function_implementation_must_have_same_name_as_decl(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::error("Function implementation name must be the same as the declaration.")
+        .with_label(span)
+}
+
+pub fn check_stmts(stmts: &oxc_allocator::Vec<'_, Statement<'_>>, ctx: &SemanticBuilder<'_>) {
+    if !ctx.source_type.is_typescript_definition() {
+        for (a, b) in stmts.iter().map(Some).chain(vec![None, None]).tuple_windows() {
```

Comment:
Walking over the statements again is not efficient, probably need some better algorithm.

Reply:
The smaller allocation `(vec![None, None])` is also not ideal.